plugins/block-dynamic-connection/src/dynamic_if.ts

```diff
@@ -11,25 +11,20 @@
 
 import * as Blockly from 'blockly/core';
 
+type DynamicIfBlock = Blockly.Block | (typeof Blockly.Blocks)['dynamic_if'];
+
 Blockly.Blocks['dynamic_if'] = {
-  /**
-   * Counter for the next input to add to this block.
-   * @type {number}
-   */
+  /** Counter for the next input to add to this block. */
   inputCounter: 1,
 
-  /**
-   * Minimum number of inputs for this block.
-   * @type {number}
-   */
+  /** Minimum number of inputs for this block. */
   minInputs: 1,
 
   /**
    * Block for if/elseif/else statements. Must have one if input.
    * Can have any number of elseif inputs and optionally one else input.
-   * @this {Blockly.Block}
    */
-  init: function() {
+  init: function(this: DynamicIfBlock): void {
```

Comment:
I think you can use method syntax here (and many places below):
```suggestion
  init(this: DynamicIfBlock): void {
```
(Note that I did not do that in my first pass on core `blocks/`, but I'm intending to do so when I go back to tidy things up.)

Reply:
Done.